From 7e5a737091056d7b897849ac044bd0ab8a4dc7b9 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Sun, 21 Jul 2019 09:02:57 +0200
Subject: [PATCH 29/39] iommu: Fix alloc_io_pgtable_ops for ARM_MALI_LPAE on
 T720

Too strict an sanity check. T720 has:

  panfrost 1800000.gpu: ias=33 oas=40

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 drivers/iommu/io-pgtable-arm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/iommu/io-pgtable-arm.c b/drivers/iommu/io-pgtable-arm.c
index 161a7d5..082efc4 100644
--- a/drivers/iommu/io-pgtable-arm.c
+++ b/drivers/iommu/io-pgtable-arm.c
@@ -1015,7 +1015,7 @@ arm_mali_lpae_alloc_pgtable(struct io_pgtable_cfg *cfg, void *cookie)
 {
 	struct io_pgtable *iop;
 
-	if (cfg->ias != 48 || cfg->oas > 40)
+	if (cfg->ias > 48 || cfg->oas > 40)
 		return NULL;
 
 	cfg->pgsize_bitmap &= (SZ_4K | SZ_2M | SZ_1G);
-- 
2.7.1

